src/caffeine/allocation_m.f90

```diff
@@ -4,11 +4,19 @@ module allocation_m
     use iso_c_binding, only: c_ptr, c_int, c_intmax_t, c_size_t, c_funptr
     implicit none
     private
-    public :: prif_allocate, prif_allocate_non_symmetric, prif_deallocate, prif_deallocate_non_symmetric
+    public :: &
+        prif_coarray_handle, &
+        prif_allocate, &
+        prif_allocate_non_symmetric, &
+        prif_deallocate, &
+        prif_deallocate_non_symmetric
 
 
-    type, public :: prif_coarray_handle
-       type(c_ptr) :: ptr
+    type :: prif_coarray_handle
+       integer(c_intmax_t), allocatable :: lcobounds(:), ucobounds(:)
+       integer(c_size_t) :: element_length
+       type(c_funptr) :: final_func
+       type(c_ptr) :: base_pointer
```

Comment:
This is almost certainly not the approach we want. 

These all need to be stored in the heap as a metadata header before the elements (as in Kate's prototyping), not in the handle itself.